bundles/org.openhab.binding.verisure/src/main/java/org/openhab/binding/verisure/internal/VerisureSession.java

```diff
@@ -155,7 +155,8 @@ public boolean unregisterDeviceStatusListener(
 
     @SuppressWarnings("unchecked")
     public boolean registerDeviceStatusListener(DeviceStatusListener<? extends VerisureThingDTO> deviceStatusListener) {
-        return deviceStatusListeners.add((DeviceStatusListener<VerisureThingDTO>) deviceStatusListener);
+        return deviceStatusListeners.contains(deviceStatusListener) ? false
```

Comment:
Perhaps it would be better to store listeners in a `Set`? This way they would be distinct by contract and lookup would be faster:
```java
Set<DeviceStatusListener<VerisureThingDTO>> deviceStatusListeners = ConcurrentHashMap.newKeySet()
```

Reply:
I like the idea of using a Set.

Reply:
@jannegpriv: would you like to propose your own PR and test it? 
I hav myself no way to test this binding.

Reply:
@lolodomo - I believe you will literally just have to replace the declaration with the suggestion above and revert the changes in `registerDeviceStatusListener`. I would consider this change at least as safe as the one above, but in any case it would be preferable to have it tested, of course.

Reply:
Done

Reply:
I'd be happy to test, if you want!